airflow/www/views.py

```diff
@@ -836,6 +839,21 @@ def log(self, session=None):
         dttm = pendulum.parse(execution_date)
         form = DateTimeForm(data={'execution_date': dttm})
         dag = dagbag.get_dag(dag_id)
+        metadata = {}
+        tailing_required = False
+        tail_lines_list = [100, 200, 500, 1000]
+        num_lines = 100
+        if conf.has_option('webserver', 'tail_lines_list'):
+            tail_lines_list = conf.get('webserver', 'tail_lines_list')
+            tail_lines_list = list(map(int, tail_lines_list.split(',')))
+        if conf.has_option('webserver', 'tail_logs'):
+            tailing_required = conf.getboolean('webserver', 'tail_logs')
+            if tailing_required and conf.has_option('webserver', 'num_lines'):
+                num_lines = conf.getint('webserver', 'num_lines')
+            metadata["num_lines"] = num_lines
```

Comment:
Please find a way of re-writing this. The nested conditions and checks make it hard to see what the intent here is. Perhaps some comment saying when/why we do things would help me understand this code.

I'm also notsure what the webserver -> tail_logs setting is for. Moving that to default config means you can add a comment to it

Reply:
I am essentially using three fields for this feature. Boolean flag `tail_logs` in `webserver` to turn on and off this feature. `tail_lines_list` in `webserver` to show default entries in dropdown and finally `num_lines` in `webserver` to pick default number of lines to tail when user lands on `get_logs` page.

Reply:
Relevant comments have been added. @ashb  please review.